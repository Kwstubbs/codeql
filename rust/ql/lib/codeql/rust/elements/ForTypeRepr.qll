// generated by codegen, do not edit
/**
 * This module provides the public class `ForTypeRepr`.
 */

private import internal.ForTypeReprImpl
import codeql.rust.elements.ForBinder
import codeql.rust.elements.TypeRepr

/**
 * A type with a higher-ranked `for` modifier. This is currently not valid Rust syntax (`for<...>` can
 * only be applied to traits to form a `TypeBound`).
 *
 * For example:
 * ```rust
 * fn foo(value: for<'a> usize) {}  // DOESN'T COMPILE
 * //            ^^^^^^^^^^^^^
 * ```
 */
final class ForTypeRepr = Impl::ForTypeRepr;
