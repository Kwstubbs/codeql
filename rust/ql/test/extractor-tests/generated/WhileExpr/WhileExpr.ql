// generated by codegen, do not edit
import codeql.rust.elements
import TestUtils

query predicate instances(
  WhileExpr x, string hasLabel__label, string hasLabel, string hasLoopBody__label,
  string hasLoopBody, string getNumberOfAttrs__label, int getNumberOfAttrs,
  string hasCondition__label, string hasCondition
) {
  toBeTested(x) and
  not x.isUnknown() and
  hasLabel__label = "hasLabel:" and
  (if x.hasLabel() then hasLabel = "yes" else hasLabel = "no") and
  hasLoopBody__label = "hasLoopBody:" and
  (if x.hasLoopBody() then hasLoopBody = "yes" else hasLoopBody = "no") and
  getNumberOfAttrs__label = "getNumberOfAttrs:" and
  getNumberOfAttrs = x.getNumberOfAttrs() and
  hasCondition__label = "hasCondition:" and
  if x.hasCondition() then hasCondition = "yes" else hasCondition = "no"
}

query predicate getLabel(WhileExpr x, Label getLabel) {
  toBeTested(x) and not x.isUnknown() and getLabel = x.getLabel()
}

query predicate getLoopBody(WhileExpr x, BlockExpr getLoopBody) {
  toBeTested(x) and not x.isUnknown() and getLoopBody = x.getLoopBody()
}

query predicate getAttr(WhileExpr x, int index, Attr getAttr) {
  toBeTested(x) and not x.isUnknown() and getAttr = x.getAttr(index)
}

query predicate getCondition(WhileExpr x, Expr getCondition) {
  toBeTested(x) and not x.isUnknown() and getCondition = x.getCondition()
}
