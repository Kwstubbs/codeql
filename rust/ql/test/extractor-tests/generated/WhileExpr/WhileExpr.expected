instances
| gen_while_expr.rs:7:5:9:5 | while ... { ... } | hasLabel: | no | hasLoopBody: | yes | getNumberOfAttrs: | 0 | hasCondition: | yes |
getLabel
getLoopBody
| gen_while_expr.rs:7:5:9:5 | while ... { ... } | gen_while_expr.rs:7:18:9:5 | { ... } |
getAttr
getCondition
| gen_while_expr.rs:7:5:9:5 | while ... { ... } | gen_while_expr.rs:7:11:7:16 | ... < ... |
